pkg/apis/policy/v1alpha1/propagation_types.go

```diff
@@ -173,6 +173,12 @@ type PropagationSpec struct {
 	// +kubebuilder:validation:Enum=Lazy
 	// +optional
 	ActivationPreference ActivationPreference `json:"activationPreference,omitempty"`
+
+	// OrphaningDeletion  represents whether to prevent removal of these managed resources
```

Comment:
I think we also need to explain each of the options - what "Never" means, what "Complete" means.

Reply:
The field is ambiguous: delete `what`?  `resource template` or `PropagationPolicy`?

I know you mean `resource template`, but it should be clearly noted(try with field name). Otherwise, user might think it works for `PropagationPolicy` since this is a specification from PropagationPolicy.

Reply:
What do you think about:
`ResourceDeletionPolicy: Background/Foreground/Orphan`, like appears on [Kubernetes Docs](https://kubernetes.io/docs/tasks/administer-cluster/use-cascading-deletion/)

When:
**Background** (default) - marks the resource from Karmada for deletion, as well as its propagated resources, and lets the GC collect the resources asynchronically.
**Foreground** - marks the resource from Karmada for deletion, as well as its propagated resources, and waits for removal of owned/propagated resources before collecting the main resource.
**Orphan** - marks the resource from Karmada for deletion, but leaves its propagated ones untouched.